src/Store/SingleTableStore.php

```diff
@@ -220,4 +271,39 @@ public function schema(): Schema
 
         return $schema;
     }
+
+    private function splitEventStream(Message $message): void
+    {
+        $toBeArchivedMessages = $this->loadNotArchived(
+            $message->aggregateClass(),
+            $message->aggregateId(),
+            $message->playhead()
+        );
+
+        foreach ($toBeArchivedMessages as $toBeArchivedMessage) {
+            if ($toBeArchivedMessage->archived()) {
+                // already archived, should we exit here since there seems something wrong with the
+                // query?
+            }
+
+            $this->archiveMessage($toBeArchivedMessage);
+        }
+    }
```

Comment:
why don't you create an update statement that simply sets the flag "archive" for all messages of an aggregate up to this playhead?

```sql
UPDATE table_name
SET archived = 1
WHERE
    aggregate_id = :aggregateId
    playhead < :playhead
    archived = 0
```

So i would not want to reload the whole stream to edit the message and then write it again.